light-blockchain/src/abstract_blockchain.rs

```diff
@@ -63,6 +63,10 @@ impl AbstractBlockchain for LightBlockchain {
             .map(|chain_info| chain_info.head.clone())
     }
 
+    fn get_genesis_hash(&self) -> Blake2bHash {
+        self.genesis_block.hash()
```

Comment:
Why is it cached for the (normal) blockchain but not for the light blockchain?

Reply:
The light blockchain stores the whole genesis block in `self`, whereas the normal blockchain does not. For the normal blockchain to query the hash each time would be resource intensive, whereas the light blockchain already has the block loaded and just fetches it's hash.

Reply:
I see. Thanks!